mne/epochs.py

```diff
@@ -1400,6 +1400,16 @@ def _get_data(self, out=True, picks=None, item=None, *, units=None,
             End time of data to get in seconds.
         %(verbose)s
         """
+        # if called with 'out=False', the call came from 'drop_bad()'
+        # if no reasons to drop, just declare epochs as good and return
+        if not out:
+            # might be BaseEpochs or Epochs, only the latter has the attribute
+            reject_by_annotation = getattr(self, 'reject_by_annotation', False)
+            if (self.reject is None and self.flat is None and
+                    self._reject_time is None and not reject_by_annotation):
```

Comment:
... one might be to check the `preload` status, and if not preloaded, use `self._get_epoch_from_raw` on the first and last indices. If these are both non-None (I think it returns `None` when it's outside the bounds?), then you know they are all non-None because the events are always monotonically increasing. Then you can `return None`. But if either is None, it's safest just to continue with the slower code path.